cfn-resources/teams/cmd/resource/team-user/team_user.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teamuser
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231001001/admin"
+)
+
+type TeamUsersAPI interface {
+	GetUserByUsername(ctx context.Context, userName string) (*atlasv2.CloudAppUser, *http.Response, error)
+	AddTeamUser(ctx context.Context, orgID string, teamID string, addUserToTeam *[]atlasv2.AddUserToTeam) (*atlasv2.PaginatedApiAppUser, *http.Response, error)
+	RemoveTeamUser(ctx context.Context, orgID string, teamID string, userID string) (*http.Response, error)
+}
+
+type TeamUsersAPIService struct {
+	mongoDBCloudUsersAPI atlasv2.MongoDBCloudUsersApi
+	teamsAPI             atlasv2.TeamsApi
+}
+
+func NewTeamUsersAPIService(client *atlasv2.APIClient) *TeamUsersAPIService {
+	return &TeamUsersAPIService{
+		mongoDBCloudUsersAPI: client.MongoDBCloudUsersApi,
+		teamsAPI:             client.TeamsApi,
+	}
+}
+
+func (s *TeamUsersAPIService) GetUserByUsername(ctx context.Context, userName string) (*atlasv2.CloudAppUser, *http.Response, error) {
+	return s.mongoDBCloudUsersAPI.GetUserByUsername(context.Background(), userName).Execute()
+}
+
+func (s *TeamUsersAPIService) AddTeamUser(ctx context.Context, orgID string, teamID string, addUserToTeam *[]atlasv2.AddUserToTeam) (*atlasv2.PaginatedApiAppUser, *http.Response, error) {
+	return s.teamsAPI.AddTeamUser(ctx, orgID, teamID, addUserToTeam).Execute()
+}
+
+func (s *TeamUsersAPIService) RemoveTeamUser(ctx context.Context, orgID string, teamID string, userID string) (*http.Response, error) {
+	return s.teamsAPI.RemoveTeamUser(ctx, orgID, teamID, userID).Execute()
+}
```

Comment:
if the signature changed, wouldn't there be the same problem just in a different file?

Reply:
if the signature changes we will have to update the resource code + the definition of the function here.

Reply:
I don't see it as a concern tbh but thanks for raising the point.

Starting from the fact that changing the signature is a breaking change (as such it's already "a problem" on its own), I would say that once you edited the signature of the client, it doesn't take much effort to update the resource as well. To me the efforts are equal.

I would be more worried if _hypothetically_ this change would introduce a risk of missing an update which would cause a runtime error, which is not the case.

Said that, separating interfaces between components (i.e. making them less coupled) is a common practice. In theory for example if one day the client would change the `RemoveTeamUser` by requiring one more mandatory field you could always leave the `resource` code as it is and only change this interface by applying default values.

Reply:
I understand that the motivation to create this API is mostly to make mocking easier. In general I would agree that it's a good idea to have an interface that we own between the service and our code. 
But in this case it can be argued that we own the SDK so I kind of agree with Andrea concern that we'll have to maintain this interface in the middle when we're also really owners of the SDK and can influence on it. So I think that it doesn't add some much value as in other cases

Reply:
> I understand that the motivation to create this API is mostly to make mocking easier

It starts because of that, yes. But in the end is not the only reason, but there is a [good answer](https://mongodb.stackenterprise.co/a/1065/747) by @gssbzn which I tend to agree with.

Reply:
> when we're also really owners of the SDK and we can influence it

The relation between different packages and teams is an interesting one (I read an article when in Meta about our tendency of evolving teams based on how the product evolves, creating weird organizational structures - I don't remember it anymore which one it is).

My short answer is that we should never assume these org structures will remain the same forever. It is always possible to think that things will evolve in a way that a CFN integration and an SDK will be managed by not-so-close teams.